src/txmempool.cpp

```diff
@@ -677,14 +677,18 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
     assert(innerUsage == cachedInnerUsage);
 }
 
-void CTxMemPool::queryHashes(vector<uint256>& vtxid)
+void CTxMemPool::queryHashes(vector<uint256>& vtxid, int64_t mintime, int64_t maxtime)
 {
     vtxid.clear();
 
     LOCK(cs);
     vtxid.reserve(mapTx.size());
-    for (indexed_transaction_set::iterator mi = mapTx.begin(); mi != mapTx.end(); ++mi)
-        vtxid.push_back(mi->GetTx().GetHash());
+    //Doesn't use the time sorted index to avoid an information leak about when a transaction entered the mempool.
```

Comment:
I'm not sure I understand the rationale behind not using the index because it would leak information. We're still querying on mi->GetTime() so 'leaking' the same information?

If the order of the result is the problem, we could just sort the result by hash before returning.